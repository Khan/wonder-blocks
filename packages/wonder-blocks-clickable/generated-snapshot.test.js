// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests:
//   1. edit the markdown documentation files in the package,
//        packages/wonder-blocks-clickable
//   2. Run `yarn run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";

// Mock react-dom as jest doesn't like findDOMNode.
jest.mock("react-dom");
import Clickable from "./components/clickable.js";

describe("wonder-blocks-clickable", () => {
    it("example 1", () => {
        const {StyleSheet} = require("aphrodite");
        const {View} = require("@khanacademy/wonder-blocks-core");
        const {MemoryRouter, Route, Switch} = require("react-router-dom");
        const Color = require("@khanacademy/wonder-blocks-color").default;
        const {Body} = require("@khanacademy/wonder-blocks-typography");

        const styles = StyleSheet.create({
            focused: {
                border: "none",
            },
            hovered: {
                textDecoration: "underline",
            },
            pressed: {
                color: Color.teal,
            },
        });

        const example = (
            <View>
                <Clickable onClick={() => alert("You clicked some text!")}>
                    {(eventState) => (
                        <Body
                            style={[
                                eventState.focused && styles.focused,
                                eventState.hovered && styles.hovered,
                                eventState.pressed && styles.pressed,
                            ]}
                        >
                            This text is clickable!
                        </Body>
                    )}
                </Clickable>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 2", () => {
        const {StyleSheet} = require("aphrodite");
        const {View} = require("@khanacademy/wonder-blocks-core");
        const Spacing = require("@khanacademy/wonder-blocks-spacing");
        const {MemoryRouter, Route, Switch} = require("react-router-dom");

        const styles = StyleSheet.create({
            row: {
                flexDirection: "row",
                alignItems: "center",
            },
            h1: {
                marginRight: "25px",
            },
        });

        // NOTE: In actual code you would use BrowserRouter instead
        const example = (
            <MemoryRouter>
                <View style={styles.row}>
                    <Clickable
                        href="/foo"
                        style={styles.h1}
                        onClick={() =>
                            console.log("I'm still on the same page!")
                        }
                    >
                        {(eventState) => <h1>Uses Client-side Nav</h1>}
                    </Clickable>
                    <Clickable href="/foo" style={styles.h1} skipClientNav>
                        {(eventState) => <h1>Avoids Client-side Nav</h1>}
                    </Clickable>
                    <Switch>
                        <Route path="/foo">
                            <View id="foo">Hello, world!</View>
                        </Route>
                    </Switch>
                </View>
            </MemoryRouter>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
