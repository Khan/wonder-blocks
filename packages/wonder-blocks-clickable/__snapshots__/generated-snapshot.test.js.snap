// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-clickable example 1 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "center",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "row",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <a
    aria-label=""
    className=""
    href="/foo"
    onBlur={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "inherit",
        "WebkitAppearance": "none",
        "WebkitFontSmoothing": "inherit",
        "background": "transparent",
        "border": "none",
        "boxSizing": "border-box",
        "color": "inherit",
        "font": "inherit",
        "lineHeight": "normal",
        "margin": 0,
        "marginRight": "25px",
        "outline": "none",
        "overflow": "visible",
        "padding": 0,
        "textDecoration": "none",
        "touchAction": "manipulation",
        "userSelect": "none",
        "width": "auto",
      }
    }
    tabIndex={0}
  >
    <h1>
      Uses Client-side Nav
    </h1>
  </a>
  <a
    aria-label=""
    className=""
    href="/foo"
    onBlur={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "inherit",
        "WebkitAppearance": "none",
        "WebkitFontSmoothing": "inherit",
        "background": "transparent",
        "border": "none",
        "boxSizing": "border-box",
        "color": "inherit",
        "font": "inherit",
        "lineHeight": "normal",
        "margin": 0,
        "marginRight": "25px",
        "outline": "none",
        "overflow": "visible",
        "padding": 0,
        "textDecoration": "none",
        "touchAction": "manipulation",
        "userSelect": "none",
        "width": "auto",
      }
    }
    tabIndex={0}
  >
    <h1>
      Avoids Client-side Nav
    </h1>
  </a>
</div>
`;
